src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/ExecutionManager/ExecutionManagerBase.ReadyToRunJitManager.cs

```diff
@@ -43,35 +47,44 @@ public override bool GetMethodInfo(RangeSection rangeSection, TargetCodePointer
             TargetPointer imageBase = rangeSection.Data.RangeBegin;
             TargetPointer relativeAddr = addr - imageBase;
 
-            int index = GetRuntimeFunctionIndexForAddress(r2rInfo, relativeAddr);
-            if (index < 0)
+            uint index;
+            if (!_runtimeFunctions.TryGetRuntimeFunctionIndexForAddress(r2rInfo.RuntimeFunctions, r2rInfo.NumRuntimeFunctions, relativeAddr, out index))
                 return false;
 
             bool featureEHFunclets = Target.ReadGlobal<byte>(Constants.Globals.FeatureEHFunclets) != 0;
             if (featureEHFunclets)
             {
-                // TODO: [cdac] Look up in hot/cold mapping lookup table and if the method is in the cold block,
-                // get the index of the associated hot block.
-                //   HotColdMappingLookupTable::LookupMappingForMethod
-                //
-                // while GetMethodDescForEntryPoint for the begin address of function at index is null
-                //   index--
+                // Look up index in hot/cold map - if the function is in the cold part, get the index of the hot part.
+                index = _hotCold.GetHotFunctionIndex(r2rInfo.NumHotColdMap, r2rInfo.HotColdMap, index);
             }
 
-            TargetPointer functionEntry = r2rInfo.RuntimeFunctions + (ulong)(index * _runtimeFunctionSize);
-            Data.RuntimeFunction function = Target.ProcessedData.GetOrAdd<Data.RuntimeFunction>(functionEntry);
-
-            // ReadyToRunInfo::GetMethodDescForEntryPointInNativeImage
-            TargetCodePointer startAddress = imageBase + function.BeginAddress;
-            TargetPointer entryPoint = CodePointerUtils.AddressFromCodePointer(startAddress, Target);
+            TargetPointer methodDesc = GetMethodDescForRuntimeFunction(r2rInfo, imageBase, index);
+            while (featureEHFunclets && methodDesc == TargetPointer.Null)
```

Comment:
Can we comment on this indexing logic? Perhaps it is already documented in the markdown?